docs/reference/search/search-your-data/run-knn-search.asciidoc

```diff
@@ -0,0 +1,248 @@
+
+* <<exact-knn,Exact, brute-force kNN>> using a `script_score` query with a
+vector function
+
+In most cases, you'll want to use approximate kNN. Approximate kNN offers lower
+latency and better support for large datasets at the cost of slower indexing and
+reduced accuracy. However, you can configure this method for higher accuracy in
+exchange for slower searches.
+
+Exact, brute-force kNN guarantees accurate results but doesn't scale well with
+large datasets. The vector function has to scan each matched document linearly,
+resulting in slow search speeds. However, you can improve latency by using a
+query to limit the number of matched documents passed to the function.
+
+[discrete]
+[[approximate-knn]]
+=== Approximate kNN
+
+To run an approximate kNN search, use the kNN search API to search a
+`dense_vector` field with indexing enabled.
+
+. Explicitly map one or more `dense_vector` fields. Approximate kNN search
+requires the following mapping options:
++
+--
+* An `index` value of `true`.
+
+* A `similarity` value. This value determines the similarity metric used to
+score documents based on similarity between the query and document vector. For a
+list of available metrics, see the <<dense-vector-similarity,`similarity`>>
+parameter documentation.
+
+include::{es-repo-dir}/mapping/types/dense-vector.asciidoc[tag=dense-vector-indexing-speed]
+
+[source,console]
+----
+PUT my-approx-knn-index
+{
+  "mappings": {
+    "properties": {
+      "my-image-vector": {
+        "type": "dense_vector",
+        "dims": 5,
+        "index": true,
+        "similarity": "l2_norm"
+      },
+      "my-tag": {
+        "type": "keyword"
+      }
+    }
+  }
+}
+----
+--
+
+. Index your data.
++
+[source,console]
+----
+POST my-approx-knn-index/_bulk?refresh=true
+{ "index": { "_id": "1" } }
+{ "my-image-vector": [230.0, 300.33, -34.8988, 15.555, -200.0], "my-tag": "cow.jpg" }
+{ "index": { "_id": "2" } }
+{ "my-image-vector": [-0.5, 100.0, -13.0, 14.8, -156.0], "my-tag": "moose.jpg" }
+{ "index": { "_id": "3" } }
+{ "my-image-vector": [0.5, 111.3, -13.0, 14.8, -156.0], "my-tag": "rabbit.jpg" }
+...
+----
+//TEST[continued]
+//TEST[s/\.\.\.//]
+
+. Run the search using the <<knn-search,kNN search API>>.
++
+[source,console]
+----
+GET my-approx-knn-index/_knn_search
+{
+  "knn": {
+    "field": "my-image-vector",
+    "query_vector": [-0.5, 90.0, -10, 14.8, -156.0],
+    "k": 10,
+    "num_candidates": 100
+  },
+  "fields": [
+    "my-image-vector",
+    "my-tag"
+  ]
+}
+----
+//TEST[continued]
+// TEST[s/"k": 10/"k": 3/]
+// TEST[s/"num_candidates": 100/"num_candidates": 3/]
+
+[discrete]
+[[tune-approximate-knn-for-speed-accuracy]]
+==== Tune approximate kNN for speed or accuracy
+
+For faster searches, the kNN search API collects `num_candidates` results from
+each shard, then merges them to find the top `k` results. This works well in
+most cases, but results are less accurate if a shard contains a disproportionate
+number of nearest neighbors.
+
```

Comment:
> This works well in most cases, but results are less accurate if a shard contains a disproportionate number of nearest neighbors.

The meaning of this sentence is not clear to me.   Why results are less accurate in this case?

I think what we should say is something like:

A higher number of `num_candidates` allows Elasticsearch to consider more candidates, which means a higher probability to find the true nearest neighbours among them, and hence higher accuracy.

Reply:
Thanks for pointing this out. Your suggestion gets to the gist of what I was intending to say here. I revised this in https://github.com/elastic/elasticsearch/pull/80857/commits/29d37f194d53c1da67de4b7376da9939803b3b08